### PR TITLE
Fix recalculating BQ around big static objects

### PR DESCRIPTION
**Pull request by wichern**
The recalculation around big static objects did not work properly.

Before this patch:
Screenshot from 2024-02-27 18-51-15
Screenshot from 2024-02-27 18-51-20
Screenshot from 2024-02-27 18-51-25

After this patch:
Screenshot from 2024-02-29 11-50-50

Maybe cause of #1516 and #1520

### CODE REVIEW

File: libs/s25main/lua/LuaWorld.cpp [hunk 1]
```diff
@@ -84,7 +84,10 @@ bool LuaWorld::AddStaticObject(int x, int y, unsigned id, unsigned file /* = 0xF
 
     gw.DestroyNO(pt, false);
     gw.SetNO(pt, new noStaticObject(pt, id, file, size));
-    gw.RecalcBQAroundPoint(pt);
+    if(2 == size)
```
Comment (Flamefire) [inline libs/s25main/lua/LuaWorld.cpp:87]: Please avoid "Yoda-conditions" --> `if(size == 2)`

**Review by Flamefire — CHANGES_REQUESTED** [review]
Maybe we can make this even better: `size >1 -> RecalcBQAroundPointBig`, `size=1 -> RecalcBQAroundPoint`, else nothing.

Could you do that?

File: libs/s25main/lua/LuaWorld.cpp (hunk 1 above)
Comment (wichern) [reply libs/s25main/lua/LuaWorld.cpp]: Thank you for your suggestions. I've updated the PR.

I have two general questions:
1. Do you use a pre-commit hook in this repo? I haven't found one.
2. Is it ok to force-push in order to keep the amount of commits low? I noticed that changes are not squashed.

Comment (Flamefire) [reply libs/s25main/lua/LuaWorld.cpp]: 1. We don't. But we do have a build target `make clangFormat` (IIRC) that is available when (the right version of) clang-format is found during configure/cmake that can be used to format (changed) files explicitely.

2. It depends ;-) If it is a single commit with a single small change then I'd say it is ok. Especially as GitHub now shows force-pushes and allows to compare the difference. Otherwise you'd better use additional commits as long as the review is in progress and rebase and squash before merge when review is finished. You can use `git commit --fixup` to squash automatically on a rebase (see the man page). But in the end having those extra commits isn't a real problem. E.g. in the case here the history would have shown the "sub-optimal" solution and the conscious change to the current solution which is often a good thing.    
So in summary: If the additional commit might be helpful (e.g. for review or future reference) or if in doubt keep it.

**Review by Flamefire — APPROVED** [review]
